etl/src/main/java/org/pih/hivmigration/etl/sql/ProgramMigrator.groovy

```diff
@@ -120,19 +120,25 @@ class ProgramMigrator extends SqlMigrator {
             # Top choice is the current treatment_status_date
             UPDATE hivmigration_programs_raw
                 SET outcome_date = treatment_status_date;
+                
             # Unless there's a regimen_outcome_date which is later than the treatment_status_date
             UPDATE hivmigration_programs_raw
                 SET outcome_date = regimen_outcome_date WHERE outcome_date IS NULL OR regimen_outcome_date > outcome_date;
-            # And if the enrollment date (which is the earlier of minimum encounter and art start date) is later
-            # than either of those, use that
-            UPDATE hivmigration_programs_raw
-                SET outcome_date = enrollment_date WHERE outcome_date IS NULL OR enrollment_date > outcome_date;
+            
+            # If no outcome date is readily available, use last visit date + 6 months as a proxy
             UPDATE hivmigration_programs_raw
                 SET outcome_date = DATE(DATE_ADD(max_visit_date, INTERVAL 6 MONTH)) WHERE outcome IS NULL;            
+            
+            # If this results in a future outcome date, then just use the current date
             UPDATE hivmigration_programs_raw
                 SET outcome_date = CURDATE() WHERE outcome_date > CURDATE();
+                
+            # Now, make sure we aren't setting an outcome date if the patient does not have an outcome
```

Comment:
This makes sense... I was wondering why we were setting outcome date for all patients, which seemed wrong?